client/src/main/java/org/apache/falcon/client/FalconClient.java

```diff
@@ -1154,13 +1152,20 @@ public APIResult updateExtensionJob(final String extensionName, final String fil
         return getResponse(APIResult.class, clientResponse);
     }
 
-    public APIResult validateExtensionJob(final String extensionName, final String filePath, final String doAsUser) {
-        InputStream entityStream = getServletInputStream(filePath);
-        ClientResponse clientResponse = new ResourceBuilder()
-                .path(ExtensionOperations.VALIDATE.path, extensionName)
-                .addQueryParam(DO_AS_OPT, doAsUser)
-                .call(ExtensionOperations.VALIDATE, entityStream);
-        return getResponse(APIResult.class, clientResponse);
+    public APIResult validateExtensionJob(final String extensionName, final String jobName,
+                                          final String configPath, final String doAsUser) {
+        String extensionType = ExtensionHandler.getExtensionType(extensionName, getExtensionJobDetail(extensionName));
+        InputStream configStream = getServletInputStream(configPath);
+        if (ExtensionType.TRUSTED.name().equalsIgnoreCase(extensionType)) {
+            ClientResponse clientResponse = new ResourceBuilder()
```

Comment:
Can you update ExtensionManager to return an exception if validate is called on custom extensions?